lib/hotel_controller.rb

```diff
@@ -0,0 +1,139 @@
+require 'Date'
+module Hotel
+  class HotelController
+    attr_accessor :reservation_array
+
+    def initialize
+      @reservation_array = []
+    end
+
+    # Wave 1
+    def rooms 
+      rooms = [1..20]
+      return rooms    # You might want to replace this method with an attr_reader
+    end
+
+
+    #NOTE the reserve_room below often works with the res_with_valid_dates function further below which validates dates
+    def reserve_room(start_date, end_date, room, block_id = nil, blocks_room_status = nil)
+      new_res = Reservation.new(start_date, end_date, room, block_id )
+      @reservation_array << new_res
+      return new_res
+    end
+
+    def reservations(date)
+      if date.class == String
+        date = Date.parse(date)
+      end
+
+      res_by_date = []
+
+      @reservation_array.each do |res| 
+        if date.between?(res.start_date,(res.end_date)-1)
+          res_by_date << res
+        end 
+      end
+
+      return res_by_date
```

Comment:
You can clean this up by using `select`:

```suggestion
      return @reservation_array.select do |res| 
        date.between?(res.start_date,(res.end_date)-1)
      end
```